server/server.py

```diff
@@ -1,34 +1,105 @@
 #!/usr/bin/env python3
 from http.server import BaseHTTPRequestHandler, HTTPServer
 from scipy.io import wavfile
+from urllib.parse import urlparse, parse_qs, unquote
 import numpy as np
 import json
+import os
 
-HOST_NAME = 'localhost'
-PORT_NUMBER = 8081
+HOST_NAME = '0.0.0.0'
+PORT_NUMBER = 8080
+DATA_DIR_PATH = '../data/speech_commands'
 
-class AudioRequestHandler(BaseHTTPRequestHandler):
-    def do_GET(self):
-        # retrieve audio to respond with
-        json_response = {}
+audio_files = {}
+
+def init_audio_files(commands, size):
+    print('initializing dataset for commands ' + str(commands) + ' with '
+          + str(size) + ' audios each')
+
+    global audio_files
+
+    if bool(audio_files):
+        audio_files = {}
+
+    # generate unknown keyword sets
+    unknown_keywords = []
+    for folder_name in os.listdir(DATA_DIR_PATH):
+        if folder_name == '_background_noise_':
+            continue
+        folder_path = os.path.join(DATA_DIR_PATH, folder_name)
+        if os.path.isdir(folder_path) and folder_name not in commands:
+            unknown_keywords.append(folder_name)
+
+    # populate positive sets
+    for folder_name in commands:
+        if folder_name == 'unknown':
+            continue
+        folder_path = os.path.join(DATA_DIR_PATH, folder_name)
+        audio_files[folder_name] = os.listdir(folder_path)[:size]
+
+    # populate negative sets
+    audio_files['unknown'] = []
+    size_per_unknown_keyword = size // len(unknown_keywords)
+    print('number of commands in unknown is ' + str(len(unknown_keywords)) +
+          ' each command with ' + str(size_per_unknown_keyword) + ' audios each')
+    for folder_name in unknown_keywords:
+        folder_path = os.path.join(DATA_DIR_PATH, folder_name)
+        audio_files['unknown'] += os.listdir(folder_path)[:size_per_unknown_keyword]
 
-        sample_rate, data = wavfile.read('../data/go/0ab3b47d_nohash_0.wav')
-        data = np.pad(data, (0, sample_rate - len(data)), 'constant')
+    # for audio_command in audio_files:
+    #     print(audio_command, len(audio_files[audio_command]))
 
-        json_response['sampleRate'] = sample_rate
-        json_response['data'] = data.tolist()
+    return
 
-        response = json.dumps(json_response)
+def get_audio(command, index):
+    file_name = audio_files[command][index]
+    print('retrieving ' + str(index) + 'th audio for ' + command + ' (' + file_name + ')')
```

Comment:
Format strings (Python 3.6) are cool:
`print(f"Retrieving {index}th audio for {command} ({file_name})")` would be equivalent and more efficient.